cmd/agent_autopilot.go

```diff
@@ -0,0 +1,122 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var agentAutopilotCmd = &cobra.Command{
+	Use:   "autopilot",
+	Short: "Background service that automatically repays FIL to pools",
+	Long:  `Background service that automatically repays FIL to pools.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+		ctx := cmd.Context()
+
+		log.Println("Starting autopilot...")
+
+		for {
+			select {
+			case <-sigs:
+				log.Println("Shutting down...")
+				Exit(0)
+			default:
+				log.Println("Checking for payments...")
+				// each loop retrieve config values aka hot-reload
+				paymentType := viper.GetString("autopilot.payment-type")
+				log.Println("Payment type: ", paymentType)
+				if paymentType == "principal" || paymentType == "custom" {
+					amount := viper.GetInt64("autopilot.amount")
+					args = append(args, fmt.Sprintf("%d", amount))
+				}
+
+				//TODO: maybe change frequency to max debt or max epoch difference
+				frequency := viper.GetFloat64("autopilot.frequency")
+
+				agent, err := getAgentAddress(cmd)
+				if err != nil {
+					log.Println(err)
+				}
+
+				account, err := PoolsSDK.Query().InfPoolGetAccount(ctx, agent)
+				if err != nil {
+					log.Println(err)
+				}
+
+				chainHeadHeight, err := PoolsSDK.Query().ChainHeight(cmd.Context())
+				if err != nil {
+					log.Println(err)
+				}
+
+				// calculate epoch frequency
+				// frequency is in days so multiply by 24 hours, 60 minutes, 60 seconds to get seconds
+				// divide by 30 seconds to get epochs
+				epochFreq := big.NewFloat(float64(frequency * 24 * 60 * 60 / 30))
```

Comment:
Should be able to use https://github.com/glifio/go-pools/blob/bcdd3dac557e3e049979532ee55c19d643b34f84/constants/constants.go#L16 here